homeassistant/components/tfa_me/config_flow.py

```diff
@@ -0,0 +1,177 @@
+            add_station_id = user_input.get(CONF_NAME_WITH_STATION_ID)
+            if not isinstance(add_station_id, bool):
+                self.name_with_station_id = False
+                errors[CONF_NAME_WITH_STATION_ID] = "invalid_name_with_station_id"
+            else:
+                self.name_with_station_id = add_station_id
+
+            # Validate the host (IP or mDNS hostname)
+            ip_host_str = user_input.get(CONF_IP_ADDRESS)
+            validator = TFAmeValidator()
+
+            # Only validate host if no previous errors exist
+            if not errors and validator.is_valid_ip_or_tfa_me(ip_host_str):
+                title_str: str = DEFAULT_STATION_NAME
+                if isinstance(ip_host_str, str):
+                    title_str = f"{DEFAULT_STATION_NAME} '{ip_host_str.upper()}'"
+
+                try:
+                    data_helper: TFAmeData = TFAmeData(self.hass, str(ip_host_str))
+                    identifier = await data_helper.get_identifier()
+
+                except TFAmeException:
+                    # Device responded or connection was attempted but failed
+                    errors["base"] = "host_empty"
+
+                except Exception:
+                    # Any unexpected exception should be logged and shown generically
+                    _LOGGER.exception(
+                        "Unexpected exception while validating TFA.me host"
+                    )
+                    errors["base"] = "unknown"
+
+                else:
+                    # Unique ID is the station identifier
+                    await self.async_set_unique_id(identifier)
+                    self._abort_if_unique_id_configured()
+
+                    # Successfully validated → create a config entry
+                    return self.async_create_entry(title=title_str, data=user_input)
+
+            elif not errors:
+                # Host is not valid at all
+                errors[CONF_IP_ADDRESS] = "invalid_ip_host"
+
+        # When user_input is None (first load) or when errors occurred,
+        # the flow must show the form again with error messages.
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA,
+            errors=errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler()
+
+
+class OptionsFlowHandler(OptionsFlow):
+    """Options flow handler (reset rain, etc.) for TFA.me integration."""
+
+    async def async_step_init(self, user_input: None) -> ConfigFlowResult:
+        """Handle options menu flow."""
+
+        # Is an option selected?
+        if user_input is not None:
+            if "select_option" in user_input:
+                if user_input["select_option"] == "action_rain":
+                    return await self.async_step_action_rain(user_input)
```

Comment:
Regardless of the comment below, I don't think we need a options menu for just a single option?